### DISCUSSION TITLE
Some basic questions about MiKTeX (upTeX/upLaTex, __doc, repository mirror)

### DISCUSSION THREAD

**Discussion by ostaith** [MiKTeX/miktex#856] [Q&A] [answered]
I like MiKTeX "Just enough TeX" concept very much, so I install MiKTeX in Windows not TeX Live. But I have some questions of MikTEX still:

1. Why MikTeX don't include upTeX (upLaTeX, eupTex...) engine? This is the best vertical typesetting engine in Windows.

2. Since most people need install doc of one package, and later MiKTeX will install `__doc` automatically, so does MiKTeX still need `Companion packages` feature, seems it will not reduce the download size of auto-installed packages too much? And this will split one package to three packages, and it's not good to manage package locally. I hope MiKTeX can return back `One Package`.

3. The last question is that, although MiKTeX have repository mirror function, but when I setup one mirror (for example in China, I setup `https://mirrors.tuna.tsinghua.edu.cn/CTAN/systems/win32/miktex/tm/packages/`), when I need install one missing package, MiKTeX will not download the missing package from mirror directly, it will query the mirror list from MiKTeX website.
This confused me too much. **Why MiKTeX can't query the package information from my setting mirror directly then install it?** 
Because in China, **query the mirror list is also very slow, even sometimes it will fail to query the mirror list** from MiKTeX website. I home this issue can be solved. @edocevoli

**Comment by edocevoli** [comment, marked as answer]
Thank you.

> Why MikTeX don't include upTeX (upLaTeX, eupTex...) engine?

Please feel free to add a ticket if you think that uptex should be a part of the MiKTeX distribution.

>  I hope MiKTeX can return back One Package

I agree that the user experience must be improved here. This will be done in the next couple of months.

> it will query the mirror list from MiKTeX website

I think there are already some tickets (maybe closed) for this feature.

**Reply by ostaith** [reply]
Thanks for your reply. I add two tickets for `question 1` #858 and `question 3` #859 already.
